source/agora/consensus/EnrollmentManager.d

```diff
@@ -711,6 +668,23 @@ public class EnrollmentManager
         return this.cycle_preimages[height];
     }
 
+    /***************************************************************************
+
+        Create signature noise for enrollment data
+
+        Returns:
+            the signature noise created using the private key
+
+    ***************************************************************************/
+
+    private Pair createSingatureNoise ()
+    {
+        Pair key_pair;
+        key_pair.v = Scalar(hashMulti(this.key_pair.v, "signature_noise"));
```

Comment:
I think this might be problematic. This means the node will always use the same starting R for every new enrollment.

@Geod24 can you offer some advice here? I think you suggested hashing the private key with the current block height, which I think would work alright.

Reply:
I agree with you. To restore the signature noise with the height you suggested, I should add a height variable where a enrollment process occurs, into the `Enrollment` structure.